### ISSUE TITLE
Cannot use table 'AspNetRoles' for entity type 'AspNetRole' since it is being used for entity type 'IdentityRole' and potentially other entity types, but there is no linking relationship.

### ISSUE THREAD

**Issue by adil-portable** [dotnet/efcore#27251]
## File a bug

I am using Db first approach using latest EF version in DOTNET 6. It does not generate AspNetUserRole table and add dictionary details as

```C#
modelBuilder.Entity<AspNetUser>(entity =>
    {
        entity.HasIndex(e => e.UserName, "UserNameIndex")
            .IsUnique();

        entity.Property(e => e.Id).HasMaxLength(128);

        entity.Property(e => e.Email).HasMaxLength(256);

        entity.Property(e => e.LockoutEndDateUtc).HasColumnType("datetime");

        entity.Property(e => e.UserName).HasMaxLength(256);


        entity.HasMany(d => d.Roles)
            .WithMany(p => p.Users);
    .UsingEntity<Dictionary<string, object>>(
        "AspNetUserRole",
        l => l.HasOne<AspNetRole>().WithMany().HasForeignKey("RoleId").HasConstraintName("FK_dbo.AspNetUserRoles_dbo.AspNetRoles_RoleId"),
        r => r.HasOne<AspNetUser>().WithMany().HasForeignKey("UserId").HasConstraintName("FK_dbo.AspNetUserRoles_dbo.AspNetUsers_UserId"),
        j =>
        {
            j.HasKey("UserId", "RoleId").HasName("PK_dbo.AspNetUserRoles");

            j.ToTable("AspNetUserRoles");

            j.HasIndex(new[] { "RoleId" }, "IX_RoleId");

            j.HasIndex(new[] { "UserId" }, "IX_UserId");

            j.IndexerProperty<string>("UserId").HasMaxLength(128);

            j.IndexerProperty<string>("RoleId").HasMaxLength(128);
        });
    });
```

Now, when I am trying to use Microsoft.AspNetCore.Identity.EntityFrameworkCore method to get user details using FindByLoginAsync method.  I get the error
![image](https://user-images.githubusercontent.com/94330648/150638152-faffae77-ebdd-4b31-b9f3-b698a648bf77.png)

My Program.cs is configured as:

![image](https://user-images.githubusercontent.com/94330648/150638185-8d4e81c9-5fd3-45f4-9e6a-c6f85a84df02.png)

I also tried by manually adding the table AspNetUserRole and adding the relationships as

![image](https://user-images.githubusercontent.com/94330648/150638432-91319959-c9a0-469d-9798-041437de6de3.png)

Please download the project from https://drive.google.com/file/d/14zmk-jijpYPbyzwOkJ63OPBMM44CHkSY/view?usp=sharing

I would be grateful if you help as ASAP. Thanks
### Include provider and version information

EF Core version: 6.0.1
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET 6.0)
Operating system:
IDE: (e.g. Visual Studio 2019 16.3)

**Comment by ajcvickers** [comment]
Duplicate of #26820. Also, this can be done now with the [EF Core Power Tools](https://github.com/ErikEJ/EFCorePowerTools),

**Comment by adil-portable** [comment]
EF 6 does not include AspNetUserRoles DbSet and is not also create DbContextDiagram. I followed this https://www.youtube.com/watch?v=uph-AGyOd8c&ab_channel=dotNET but it is for EF 5. Please kindly help me ASAP.

Can I copy paste AspNetUserRoles  class generated using EF 5 in EF 6 ? to make it work?
Thanks

**Comment by adil-portable** [comment]
Even by using EF power tool for EF 6 I get AspNetUserRole relationship like 
![image](https://user-images.githubusercontent.com/94330648/152096829-d24159c1-38ed-4e6c-8855-90482802c9cb.png)
Goal is to get Role by user
![image](https://user-images.githubusercontent.com/94330648/152096976-395b1c41-cb6a-4e6e-9c03-b4f604448308.png)
DbContext is 
![image](https://user-images.githubusercontent.com/94330648/152097066-f8d5e674-94b1-4b00-a369-a4de30f7eae7.png)
I am inheriting 
![image](https://user-images.githubusercontent.com/94330648/152097126-14b52159-d7f4-4d7a-aaef-e92b3cf64323.png)

**Comment by ErikEJ** [comment]
@adil-portable use the advanced option to actually generate an entity

**Comment by adil-portable** [comment]
Thank you very much @ErikEJ I am now able to generate AspNetUserRole object but I am still getting the same error:

**Comment by grahamoriley** [comment]
I am getting the same error as you: [adil-portable](https://github.com/adil-portable) in a web .net 6 api.

Very odd. No fix yet!

Thanks
Graham

**Comment by a-kanaan** [comment]
same problem, any solution please?

**Comment by anupamtdab** [comment]
Has anybody got the solution please?

**Comment by jorgearana** [comment]
I have the same, i guess the solution is to forget .Net, there is no answers

**Comment by blessedneelesh** [comment]
I am getting same error like @jorgearana 

Any solution please?

**Comment by Firefly-night-SEA** [comment]
I have a fix that works for me

First inherits identity db like this
public partial class YourDbContext : IdentityDbContext<IdentityUser, IdentityRole, string>

Then delete the declaration of identity models inside of YourDbContext.cs, leaving only the fields you added.

Inside of OnModelCreating, change the identity tables into this:

        modelBuilder.Entity<IdentityRole>(entity =>
        {
            entity.HasIndex(e => e.NormalizedName, "RoleNameIndex")
                .IsUnique()
                .HasFilter("([NormalizedName] IS NOT NULL)");
        });

        modelBuilder.Entity<IdentityUser>(entity =>
        {
            entity.HasIndex(e => e.NormalizedUserName, "UserNameIndex")
                .IsUnique()
                .HasFilter("([NormalizedUserName] IS NOT NULL)");
        });

        modelBuilder.Entity<IdentityUserRole<string>>(entity =>
        {
            entity.HasKey(r => new { r.UserId, r.RoleId });
            entity.ToTable("AspNetUserRoles");
            entity.HasIndex(r => r.RoleId, "IX_AspNetUserRoles_RoleId");
        });

Basically, I changed AspNetUser and AspNetRole inside of OnModelCreating into IdentityUser, etc...

**Comment by jorgearana** [comment]
Thanks, i will try it.

Regards

Ing. Jorge A. Arana Neyra
980 599 023
www.jorgearananeyra.com<http://www.jorgearananeyra.com>
"Bienvenido a la nube, programas y datos que pueden ser vistos por todos sin gastar en servidores "
________________________________
De: Firefly-night-SEA ***@***.***>
Enviado: domingo, 9 de febrero de 2025 17:35
Para: dotnet/efcore ***@***.***>
Cc: Joge Arana ***@***.***>; Mention ***@***.***>
Asunto: Re: [dotnet/efcore] Cannot use table 'AspNetRoles' for entity type 'AspNetRole' since it is being used for entity type 'IdentityRole' and potentially other entity types, but there is no linking relationship. (Issue #27251)


I have a fix that works for me

First inherits identity db like this
public partial class YourDbContext : IdentityDbContext<IdentityUser, IdentityRole, string>

Then delete the declaration of identity models inside of YourDbContext.cs, leaving only the fields you added.

Inside of OnModelCreating, change the identity tables into this:

    modelBuilder.Entity<IdentityRole>(entity =>
    {
        entity.HasIndex(e => e.NormalizedName, "RoleNameIndex")
            .IsUnique()
            .HasFilter("([NormalizedName] IS NOT NULL)");
    });

    modelBuilder.Entity<IdentityUser>(entity =>
    {
        entity.HasIndex(e => e.NormalizedUserName, "UserNameIndex")
            .IsUnique()
            .HasFilter("([NormalizedUserName] IS NOT NULL)");
    });

    modelBuilder.Entity<IdentityUserRole<string>>(entity =>
    {
        entity.HasKey(r => new { r.UserId, r.RoleId });
        entity.ToTable("AspNetUserRoles");
        entity.HasIndex(r => r.RoleId, "IX_AspNetUserRoles_RoleId");
    });


Basically, I changed AspNetUser and AspNetRole inside of OnModelCreating into IdentityUser, etc...
